components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/v1/impl/SystemScopesApiServiceImpl.java

```diff
@@ -71,4 +80,47 @@ public Response systemScopesGet(MessageContext messageContext) throws APIManagem
         }
         return null;
     }
+
+    public Response updateRolesForScope(ScopeListDTO body, MessageContext messageContext) throws APIManagementException {
+        String tenantDomain = MultitenantUtils.getTenantDomain(RestApiUtil.getLoggedInUsername());
+        //read from tenant-conf.json
+        JsonObject existingTenantConfObject = new JsonObject();
+        try {
+            APIMRegistryService apimRegistryService = new APIMRegistryServiceImpl();
+            String existingTenantConf = apimRegistryService.getConfigRegistryResourceContent(tenantDomain,
+                    APIConstants.API_TENANT_CONF_LOCATION);
+             existingTenantConfObject = new JsonParser().parse(existingTenantConf).getAsJsonObject();
+        } catch (RegistryException | UserStoreException e) {
+            APIUtil.handleException("Couldn't read tenant configuration from tenant registry", e);
+        }
+        try {
+
+            JSONObject responseJson = new JSONObject();
+            JSONArray scopeJson = new JSONArray();
+
+            for(int i = 0; i < body.getList().size(); i++) {
+                JSONObject scopeRoleJson = new JSONObject();
+                scopeRoleJson.put("Name", body.getList().get(i).getName());
+                scopeRoleJson.put("Roles", body.getList().get(i).getRoles());
+                scopeJson.put(scopeRoleJson);
+            }
+
+            responseJson.put("Scope", scopeJson);
+            //Here we are removing RESTAPIScopes from the existing tenant-conf
+            // Adding new RESTAPIScopes to the existing tenant-conf.
+            existingTenantConfObject.remove("RESTAPIScopes");
+            Gson gson = new Gson();
+            existingTenantConfObject.add("RESTAPIScopes",
+                    gson.fromJson(responseJson.toString(), JsonElement.class));
```

Comment:
Can't we do this without using gson library?

Reply:
@Meruja have a look at https://github.com/wso2/carbon-apimgt/blob/0f3201124b8376c75c5597c7eb718d585773cb26/components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java#L7386-L7391

Reply:
Fixed with bb70f77